contracts/ui/UiPoolDataProvider.sol

```
@@ -293,35 +294,46 @@ contract UiPoolDataProvider is IUiPoolDataProvider {
+                provider.getPriceOracle()
+            );
+            address sourceAddress = oracle.getSourceOfAsset(lpTokenAddress);
+            if (sourceAddress == address(0)) {
+                return lpTokenInfo;
+            }
+            source = IUniswapV3OracleWrapper(sourceAddress);
+        }
 
-            (
-                lpTokenInfo.token0,
-                lpTokenInfo.token1,
-                lpTokenInfo.feeRate,
-                lpTokenInfo.positionTickLower,
-                lpTokenInfo.positionTickUpper,
-                lpTokenInfo.currentTick
-            ) = source.getPositionBaseInfo(tokenId);
+        //try to catch invalid tokenId
+        try source.getTokenPrice(tokenId) returns (uint256 tokenPrice) {
+            lpTokenInfo.tokenPrice = tokenPrice;
+
+            UinswapV3PositionData memory positionData = source
+                .getOnchainPositionData(tokenId);
+            lpTokenInfo.token0 = positionData.token0;
+            lpTokenInfo.token1 = positionData.token1;
+            lpTokenInfo.feeRate = positionData.fee;
+            lpTokenInfo.liquidity = positionData.liquidity;
+            lpTokenInfo.positionTickLower = positionData.tickLower;
+            lpTokenInfo.positionTickUpper = positionData.tickUpper;
+            lpTokenInfo.currentTick = positionData.currentTick;
 
             (
                 lpTokenInfo.liquidityToken0Amount,
                 lpTokenInfo.liquidityToken1Amount
-            ) = source.getLiquidityAmount(tokenId);
+            ) = source.getLiquidityAmountFromPositionData(positionData);
 
             (
                 lpTokenInfo.lpFeeToken0Amount,
                 lpTokenInfo.lpFeeToken1Amount
-            ) = source.getLpFeeAmount(tokenId);
+            ) = source.getLpFeeAmountFromPositionData(positionData);
 
             lpTokenInfo.baseLTVasCollateral = 7500;
             lpTokenInfo.reserveLiquidationThreshold = 8000;
-        }
+        } catch {}
```

should we throw any errors here?

---

yeah. but frontend team ask me to suppress all kinds of errors.